android/expoview/src/main/java/host/exp/exponent/kernel/Kernel.java

```diff
@@ -493,6 +508,53 @@ public void handleIntent(Activity activity, Intent intent) {
       }
     }
 
+    openDefaultUrl();
+  }
+
+  private void openExperienceFromNotificationIntent(Intent intent) {
+    NotificationResponse response = unmarshalleNotificationResponse(intent.getByteArrayExtra(NOTIFICATION_RESPONSE_KEY));
+    if (response == null || response.getNotification() == null || response.getNotification().getNotificationRequest() == null) {
+      openDefaultUrl();
+      return;
+    }
+
+    String experienceIdString = ScopedNotificationsUtils.getExperienceId(response);
```

Comment:
Hm, if we use `ScopedNotificationsUtils` from kernel we should probably move it to unversioned part of `expoview` so it… doesn't get versioned. (Otherwise kernel would use only the latest version of `ScopedNotificationsUtils` making it more difficult to maintain with multiple SDKs.)